include/rmm/cuda_stream_view.hpp

```diff
@@ -64,26 +64,12 @@ class cuda_stream_view {
   /**
    * @brief Return true if the wrapped stream is the CUDA per-thread default stream.
    */
-  bool is_per_thread_default() const noexcept
-  {
-#ifdef CUDA_API_PER_THREAD_DEFAULT_STREAM
-    return value() == cudaStreamPerThread || value() == 0;
-#else
-    return value() == cudaStreamPerThread;
-#endif
-  }
+  [[nodiscard]] inline bool is_per_thread_default() const noexcept;
```

Comment:
What rule prompted moving the definition of these functions outside of the class?

Reply:
A bit complicated:

`cudaStreamPerThread` is defined using a C-style cast (yay!), something like `(cudaStream_t)0x2`.  To avoid that I have to NOLINT, and if I did it inside this function where it was, I would have to NOLINT two places.

But I already had `static cuda_stream_view cuda_stream_per_thread{cudaStreamPerThread};`, which has to be defined after the class is defined. So by moving the implementation of these functions outside the class, I can use those static variables, and I only have to NOLINT the static variables.

It's one of those "6 of one, half dozen of the other" situations. I can change it back and NOLINT here instead if you prefer.